### PR TITLE
Added check for belief.type

### PR DESCRIPTION
**Pull request by Why-not-now**
Added a check for belief.type for None and Any. Currently both crash the game when trying to use `Gain a free [Any] belief` unique.

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/Belief.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Belief() : RulesetObject() {
         else UniqueTarget.FollowerBelief
 
     override fun makeLink() = "Belief/$name"
-    override fun getCivilopediaTextHeader() = FormattedLine(name, icon = makeLink(), header = 2, color = if (type == BeliefType.None) "#e34a2b" else "")
+    override fun getCivilopediaTextHeader() = FormattedLine(name, icon = makeLink(), header = 2, color = "")
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/Belief.kt:23]: Why this change?

Comment (Why-not-now) [reply core/src/com/unciv/models/ruleset/Belief.kt:23]: I thought it was not intended to have these 2 types available to modders, especially when it crashes if you try to use `Gain a free [Any] belief`.

Comment (Why-not-now) [reply core/src/com/unciv/models/ruleset/Belief.kt:23]: **Platform:** Desktop
**Version:** 4.9.6 (Build 941)
**Rulesets:** [Civ V - Vanilla, Brave New World, Upside Down, Medieval Civilizations, Community Maps, Barbarian xp farm, Sandbox, Latin American_Civs, Ancient Civilizations, Civ V - Gods & Kings, test_mod_unciv, 5Hex Tileset, Combined, Difficulty_Sandbox]
**Last Screen:** `com.unciv.ui.screens.pickerscreens.ReligiousBeliefsPickerScreen`

--------------------------------

OS: Windows 10
Java: Oracle Corporation 25.391-b13
	Max Memory: 3602 MB
System default encoding: windows-1252


--------------------------------


**Message:**
```
java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(Unknown Source)
	at com.badlogic.gdx.graphics.Color.valueOf(Color.java:344)
	at com.badlogic.gdx.graphics.Color.valueOf(Color.java:338)
	at com.unciv.ui.screens.pickerscreens.ReligionPickerScreenCommon.getBeliefButton(ReligionPickerScreenCommon.kt:83)
	at com.unciv.ui.screens.pickerscreens.ReligionPickerScreenCommon.getBeliefButton$default(ReligionPickerScreenCommon.kt:73)
	at com.unciv.ui.screens.pickerscreens.ReligiousBeliefsPickerScreen.loadRightTable(ReligiousBeliefsPickerScreen.kt:206)
	at com.unciv.ui.screens.pickerscreens.ReligiousBeliefsPickerScreen.access$loadRightTable(ReligiousBeliefsPickerScreen.kt:28)
	at com.unciv.ui.screens.pickerscreens.ReligiousBeliefsPickerScreen$addChoosableBeliefButton$1.invoke(ReligiousBeliefsPickerScreen.kt:267)
	at com.unciv.ui.screens.pickerscreens.ReligiousBeliefsPickerScreen$addChoosableBeliefButton$1.invoke(ReligiousBeliefsPickerScreen.kt:265)
	at com.unciv.ui.screens.pickerscreens.ReligionPickerScreenCommon$onClickSelect$1.invoke(ReligionPickerScreenCommon.kt:101)
	at com.unciv.ui.screens.pickerscreens.ReligionPickerScreenCommon$onClickSelect$1.invoke(ReligionPickerScreenCommon.kt:99)
	at com.unciv.ui.components.input.ActivationActionMap.activate(ActivationActionMap.kt:56)
	at com.unciv.ui.components.input.ActorAttachments.activate(ActorAttachments.kt:42)
	at com.unciv.ui.components.input.ActivationExtensionsKt.activate(ActivationExtensions.kt:17)
	at com.unciv.ui.components.input.ActivationListener.tap(ActivationListener.kt:15)
	at com.badlogic.gdx.scenes.scene2d.utils.ActorGestureListener$1.tap(ActorGestureListener.java:52)
	at com.badlogic.gdx.input.GestureDetector.touchUp(GestureDetector.java:206)
	at com.badlogic.gdx.scenes.scene2d.utils.ActorGestureListener.handle(ActorGestureListener.java:125)
	at com.badlogic.gdx.scenes.scene2d.Stage.touchUp(Stage.java:354)
	at com.unciv.ui.screens.basescreen.UncivStage.access$touchUp$s80204510(UncivStage.kt:17)
	at com.unciv.ui.screens.basescreen.UncivStage$touchUp$1.invoke(UncivStage.kt:87)
	at com.unciv.ui.screens.basescreen.UncivStage$touchUp$1.invoke(UncivStage.kt:87)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
	at com.unciv.ui.screens.basescreen.UncivStage.touchUp(UncivStage.kt:87)
	at com.badlogic.gdx.InputEventQueue.drain(InputEventQueue.java:70)
	at com.badlogic.gdx.backends.lwjgl3.DefaultLwjgl3Input.update(DefaultLwjgl3Input.java:189)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Window.update(Lwjgl3Window.java:378)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:193)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:167)
	at com.unciv.app.desktop.HardenGdxAudio.<init>(HardenGdxAudio.kt:46)
	at com.unciv.app.desktop.DesktopLauncher.main(DesktopLauncher.kt:76)

```
**Save Mods:**
```
[Civ V - Gods & Kings, test_mod_unciv]
```

**Save Data:**
<details><summary>Show Saved Game</summary>

```
H4sIAAAAAAAAANVdW3fbuBH+Kzx8aXvWTEXd5Tdbm4vb2FZtt3nI+uRAJCShIQEtSNpWcvLfOwOSkpxEjqxI325fbBIEMfPNDYMBRX7276TNlNH+8WdfF+lYWv+4deRHVopcxu9UPuMruXzI/WO//WLwousfLTsO2uGXL9RZ3alEfRI5jZP5x+8/c8uFSCXdcirsWFgl6MIRDRPxcHTL3CQqUjIrz6j7kG7NbRFVY3Dr74XM8nOhxZRpcYuViZpSh0eNU5PEUp9M68GmzPpImnkiywaZzekeuuHRbXcqyo1dPB6/UDp71JIL4igSyclZNdbDPDFWxldyWgqN8H/254lYSHuzmDPiN0UqtH+0JoOTRMl1/O5/diUzKWw0kzGJzD+ZWhUVSV5Y6d+WPbMbU/fhHiOT59IuaJgTrVKReG+KbCx0XDbxQO7oXGmlp3RwLUgp7uidVXl5dGPFfF4dzqT3biZlwveIzGh396k1+hNdMPZj2W0oEqljYenwck6iYBhvjM3kWEQfvSsVV93W1OcGzGcyFVX/l3qqtJS27PqrFanwiHFvZGTuqA4La6WO+PCMGFgjP5qpxGRmPnPd2M68a2lJd5LOXxcqiZnAucxJHkOR5UtsJjFTd89M3YmklIyIZsSHOx7NFlnFXFxEomJ7aNK5yLj1Oi8lMyKmeVBvZKW7cCp0xdpJZIqsQnhC4I026cJxpefmPpYsspcRt1Z9SEWK9OpUfORfiDs1rQk7AElhK5ZlqrJSMtfkJMTAREUeg1rqWiwBnumY2CAPqz2Q2q7UpFL9OdlALuzCcwM5qb2SNmd3LRlMiCGyPJU77at6VMIvUlLQvev160KLlHjnoeXEKnKPmtQr8shZ7i6QF0RSjBPpjYTNGfKViJVx/1VijXC6KikaXQpllIhahqckAVWfkCbGLF1H4yRnGa4EMJoJmxKtwrlmRciZKN9GKo69E5tmpfxrOyDVFuQ/rreJPlaWd1FECXmY90plWUnsRhKDJk0L4rCKaEf+W5HVt45NrfP4TpBEY+8R49cUNuk0l84uzViROG5cEOGG14kZryuKxZxQiF1jpKj4YBmqiIW5NNULoQ0HBl2DelWwMXk3HFZuH0dVP5uZIokv51KPuHUxUgSaohp5qfypkLtsus4JK4W3V6bQHAauqnZibGKlPKVTOXG8jISmeFBOMlHCHnbs/1fciReJ0NMXZzqXU2dn5KkFjRh+4Ri32Lo3dZ9UPAzJkM8ooPqiK9pRbxwGYV8Mgna/3QoGzbAXxI0xXeq12qGIaYxSSsOZMZmsuSwZX0rqoPPL3JoHlbL3EdhTMV4kLCX/wmjpf2f6If1zLFifaUkHU3aLY4pVJl5G4GrG5kjqzUTmZRQFaEL3KLq+50Zpb6mXispxfNblGZ/8fW0UmodEtCLFsj0pr/C8d/vlls0od/ZGl9UzxK7LiZHnoDfqjuPKSoF36zPm3MyLRNQ0Hb3F13brj2keyE95MqABHJyR4FjEAKK1zv8qJBvMe78UgHMZmtApN1mOX9v2V5qtkgFyZoeV5/wH/zgIj/wF/6vOy1N3tmyuGx5W/+kCMXbPPMQ39ZCPx6PrKrukQKu0SIZiTjE8KQ3SIdIUQuleU11h5RXsJdzBX6llWCZxjrFvEhc/N/O3cpLzIdFuOtJtQkkBjuLtlQvr5bWgvBi02dM4YF9Lqc9Sst07mdL0tO6oFDpfFJqiy4v/Uk7x4o3IZudi/h/JztAkhVJ3W1pMKcZvHfw6rxKFyr+d3XkR3Ugq+3LLSiODfkl56+U9hfrvSsmNvJaDln61fQL2f5ShPoogld9sDiDfyV/JXMarJN31itVk4oTBYeVa5nniAi7bPymTCaViThMUiddNqbx2KEV9vvC0vPfoMis7LzPiIc3kJqVzar6mxMNNIpQasM2GDXIFFfMyo0nrjpks7S7srIKEoqmAmEpN7Hybptb8A518cGbGPp5JZ+S9RtjstLqDTrPTqbz1Lc3LzrvGNIHfSGuF4vD62pLRkbFxKKolcrpwVlAKkP1v3b4f2SBpgXIZNj67HqpmRMsF4s9+Y20oSj2cPZHbUiLpRmu4NYPJVBXXvoolOzD7PIYutmFoGcy+YmdoKADslZWvif95xPAHCeDhe2T/IDP4k2hibRZq7sTQivaPKZWEvqXzI82v0XjCww/Pf7iB/0tanund+Acwvm+Wl4LY0YSfI/Fg12j5NP+HNxWYNwF00IRpu1VSaj1J6TkjlgN+O15tPj+ILM1989PcwE/tjttFuv3xE+7Ozx4Z+QmRbBX7d4g4myepH0SWbXxlB+/b34D7M569++kepdbatx5aG8ziZznc44jtuprw1IiPl18n2tXBPbf8fSa99qqyAaBWywtFr7UB3fNmDhS3zQ3cPm9eQXEb7s4tjM2fUP5Ws/QePWPTJP6sGevpOLRHbjctAg/pxj+YB/YepVDkYMEXO6fAtNXGWkcbavo1OhS99oYwdFh0MHqdkl4HRK8kh6JWSxNFr41F18Kia21A97zcEcVt8+e4DbHchrtzC2MTaNrgzD3YlLof0jRxMX5Tee6guSksH9imhLHvEI/LvLGpI2zqwuYbMG11sNbRwWbeHaxnd7CZdwcblTvYOmMHPKN2S3pdEL2SHIpaLU0UvQ4WXRuLro1F18Kia2HRNbHomlh0IRZdiEMHgwVUF3gNHWxaRB90FQebUTdthh10FQfLvjY90nLQVRy4Zo1NnHFrVOwiCzbZYHNXmLa6WOvoYteoXaxnd7Fr1C42Knexa9QudkbtYndruuBsqFfS64HoleSepraLvFD8d7fgf/8VBBS9DhZdG4uujUXXwqJrYdE1seiaWHQhFl2IQweDBVQXuN4TAAs+qwoCLCPY9EjEQSsI4CepsY/04eoj2MfQwDt9AXYjE7wCDrALfNhUil1ZwLTVw1pHD1sf6WE9u4etj/SwUbmHrY/0sDNqD1sf6WGzoR5257IHzmT7Jb0+iF5J7mlqj39LXb9l53Y3QaKA9bYAtktZC8V/dwv+91/WQtHrYNG1sejaWHQtLLoWFl0Ti66JRRdi0YU4dDBYQHWBi5ABsAq5KmuBH3nDPs+HK9pt+tXVQctauF/gQldQ4Ee8A+wT7OBHHwLskx3gskyArTrBEgXsghCmrT7WOvrYol0f69l9bNGuj43KfWzRro+dUfvYol0fmw31sUW7PjaT7e/7Eao+eJ0xKOkNQPRKck9T20/RsY8F1ocB62GB9bYAtks1FcV/dwv+919NRdHrYNG1sejaWHQtLLoWFl0Ti66JRRdi0YU4dDBYQHWBa98BsPi9qqaCH//FPtuMqxVjf40Nfv1YgH3/GPqVEtg3ZoB/ahVgf0kGfgwswD7lBq4GBthiJywNwq7jYdoaYK1jgK0VD7CePcDWigfYqDzA1ooH2Bl1gK0VD7DZ0ABbKx5gM9nBvmvFA+w6YwB7YHSAXh42qvmlcSiKafWV5n9rlTNVU305bfXhSlN95fLymyvVVxLfCWuV4Y8TRu5j2vn5kpem+0hkahhT9YHHtLp4LlNj6w+4fr0KL2ET6i/u62zuI+9C/wSPqy9JHojHjVdhRnk4G/l+ioWA1gdD6+Og9cDQettA22W/BIaguw2C/e+YwAh2wPjaYHxtML4WGF8LjK8JxtcE4wvB+EIgPhwwpMrAe10BcLNrtXsC/pUJ9ic0uL0h7JtowK8hDrDvIQZ/eCfAfnkH/Cq3APumOvDPzAPsr+jBTxsH2IepwdX/ALu5Acz0YAlKA7s9tKzfYfHB3K3GhyJY0cN9N7SBjc4VPdy3qhrYubWih3uLewObGVX0cO+Ya2Dz2oreHreKagSwtwA2YJtFNTTYL8EbsEf9a2iHWXoTN1kuLDVM35pIVBsz73mbZSpSORKW/ubS8m6NH6vJREVFki8ebfNkcyljajmnzvnMbQVZ4sqkF0U6lvZyMkrEwg2R20Ie+anS31yhBXgqHr7XPK+P33/2o5nJpB6qO8aYEEQGV3a4Wcx58+lNkQrtsxzX+9ayZ7i6ojFU+eI6FzlvOZGIpRbjRF6VfJdyGBmT1Dzbby+8909SaVUkiIcTK8bKHRRZbsujT7mMmL/VBtnp4pMg2RcpHQ9J6DMxlXwok5w7DmdK852/Sp0K+5GOXk4Xc/6M5Es9rfT8Mp8pM3cEXhFPEd//WloCveAjK6VrOtOOrTMdu65n1vxeGMVE/iHm7qvm/yTL5EvnRk9N4nqNSMzlgUkWWpbHVyblAa+KrDy/VoLZv6bbZkLx0Zytkf7fy1jywc1Mem8Kpxo+PBcLUR1eyHwmLSOpL17muSHeM/YKK6ZkhafCjgVJUC/N5U5FbLysYLYC/0YRR7e8cxi7c9Jg/oFOPhRk9Xfuini4KSyPEHYapFsRRXKey/jcxMOZjD6+tNawRfF2qiaax97jMY69v74ySWLupT2VZGaTv3k31MG7MFr+JfMKrX4vpPeb//6X0HttkvjI+6XpjayJi4jt49abGOtJEc28aWLGIvEisjVq5CGJnEc6zDxLI0+p92++R2fa5J5wJGPPaE/lmUduOZW5lxPu3/QzOT3Riz+EUf/Lcte39OF1R+WAcsaBQna6k2jQi4No0JVBeyxlMOjLbhCLcBI1wkZv0I9WUeua4xPrk+PV/wBvd6xrypwAAA==
```
</details>

Beliefs.json:

```json
[
    {
        "name": "Test None",
        "type": "None",
        "uniques": ["[+1 Gold, +2 Production] for each global city following this religion", "Provides [100] [Uranium]"]
    },
    {
        "name": "Test Any",
        "type": "Any",
        "uniques": ["[+1 Gold, +2 Production] for each global city following this religion", "Provides [100] [Uranium]"]
    }
]
```

Comment (Why-not-now) [reply core/src/com/unciv/models/ruleset/Belief.kt]: I see now. And I agree ketchup do run with religions.

**Review by yairm210 — APPROVED** [review]
